pkg/proxy/ipvs/proxier.go

```diff
@@ -1493,53 +1496,71 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 		newEndpoints.Insert(epInfo.String())
 	}
 
-	if !curEndpoints.Equal(newEndpoints) {
-		// Create new endpoints
-		for _, ep := range newEndpoints.Difference(curEndpoints).UnsortedList() {
-			ip, port, err := net.SplitHostPort(ep)
-			if err != nil {
-				glog.Errorf("Failed to parse endpoint: %v, error: %v", ep, err)
+	// Create new endpoints
+	for _, ep := range newEndpoints.List() {
+		if curEndpoints.Has(ep) {
+			// check if newEndpoint is in gracefulDelete list, is true, delete this ep immediately
+			uniqueRS := vs.String() + "/" + ep
+			if !proxier.gracefuldeleteManager.RSInGracefulDelete(uniqueRS) {
 				continue
 			}
-			portNum, err := strconv.Atoi(port)
+			glog.V(5).Infof("new ep %q is in graceful delete list", uniqueRS)
+			err := proxier.gracefuldeleteManager.DeleteRsImmediately(uniqueRS)
```

Comment:
Won't this drop existing connections if an endpoint is flapping? For example, if a pod is under load and goes NotReady temporarily. I don't think we want to drop connections in this case. It should just remove it from the queue rather than deleting the real server.

Reply:
That's a good point, thanks. Update to weight=1 would be better.
But is there another situation, that both service and pod have been deleted, and user created another service is pod, match the original vs/rs in the delete list. 
Just little worry about this case, seems like doesn't matter.